src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -1250,7 +1250,7 @@ public ChainLevelInfo FindLevel(long number)
 
         public Keccak HeadHash => Head?.Hash;
         public Keccak GenesisHash => Genesis?.Hash;
-        public Keccak PendingHash => BestSuggestedHeader?.Hash;
+        public Keccak PendingHash => Head?.Hash;
```

Comment:
I might prefer changing:

public Block FindPendingBlock() => FindBlock(PendingHash, BlockTreeLookupOptions.None);
public BlockHeader FindPendingHeader() => FindHeader(PendingHash, BlockTreeLookupOptions.None);

in IBlockFinder, what do you think?

Reply:
I was thinking of both approaches and decided that the one that you are pointing at is at higher risk of introducing inconsistencies
The was I am changing it ensures that pending will have the same everywhere in the system